src/clinical/clinical-service.ts

```diff
@@ -324,14 +334,69 @@ export const getDonorSubmissionErrorUpdates = async (
     submitterDonorIds: [],
   };
 
-  await donorDao.findByPaginatedProgramId(programId, errorQuery).then(donorData => {
-    validDonors = donorData.donors
-      .filter(donor => donor.schemaMetadata.isValid)
-      .map(({ donorId }) => Number(donorId));
-  });
+  const donorData = (await donorDao.findByPaginatedProgramId(programId, errorQuery)).donors;
+
+  const validDonorIds = donorData
+    .filter(donor => donor.schemaMetadata.isValid)
+    .map(({ donorId }) => Number(donorId));
+
+  const clinicalErrors: ClinicalErrorsResponseRecord[] = [];
+
+  const invalidDonorIds = errorDonorIds
+    .filter(donorId => !validDonorIds.includes(donorId))
+    .filter((entity, index, array) => array.indexOf(entity) === index);
+
+  for (const donorId of invalidDonorIds) {
+    const currentDonor = donorData.filter(donor => donor.donorId === donorId)[0];
+    const { submitterId: submitterDonorId } = currentDonor;
+    const currentDonorErrors = clinicalMigrationErrors
+      .filter(errorRecord => errorRecord.donorId === donorId)
+      .map(migrationError => migrationError.errors)
+      .flat();
+
+    const currentDonorEntities = currentDonorErrors
+      .map(error => error.entityName)
+      .filter((entity, index, array) => array.indexOf(entity) === index);
+
+    const migrationVersion =
+      lastMigration?.toVersion || (await dictionaryManager.instance().getCurrentVersion());
+    const schemaName = await dictionaryManager.instance().getCurrentName();
+
+    const migrationDictionary = await dictionaryManager
+      .instance()
+      .loadSchemaByVersion(schemaName, migrationVersion);
+
+    currentDonorEntities.forEach(entityName => {
+      const entityValidationErrors: readonly DeepReadonly<
```

Comment:
`readonly DeepReadonly<...>` is redundant.
`readonly` being the TS utility function and `DeepReadonly` being `deep-freeze` npm package used throughout the project

personally I think using TS directly is the better approach moving forward, but there's a lot of legacy with `DeepReadonly` which make impact this decision.

Reply:
Migration Manager `validateDonorEntityAgainstNewSchema` returns `readonly DeepReadonly<dictionaryEntities.SchemaValidationError>[]`

I added this assertion to appease TS linter but it looks like that is no longer necessary, was likely from earlier stage of development
Pushed commit to remove it + use implicit types

I would love to simplify `readonly DeepReadonly` if possible